File: 20-valid-parentheses/20-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
  if (s.length % 2 !== 0) {
    return false;
  } else {
    const leftSymbol = [];
    for (let i = 0; i < s.length; i++) {
      if (s[i] === '{' || s[i] === '(' || s[i] === '[') {
        leftSymbol.push(s[i]);
      } else if (
        (s[i] === '}' && leftSymbol.length !== 0 && leftSymbol.slice(-1)[0] === '{') ||
        (s[i] === ']' && leftSymbol.length !== 0 && leftSymbol.slice(-1)[0] === '[') ||
        (s[i] === ')' && leftSymbol.length !== 0 && leftSymbol.slice(-1)[0] === '(')
      ) {
        leftSymbol.pop();
      }else{
          return false;
      }
    }
      return !leftSymbol.length;
  }
};

File: 20-valid-parentheses/20-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        if(s.size() % 2 != 0){
            return false;
        }
        vector<char> leftSymbol;
        for(int i = 0; i <s.size(); i++){
          if(s[i] == '(' || s[i] =='[' || s[i] == '{'){
              leftSymbol.push_back(s[i]);
          }else if(
(s[i] == ')' && leftSymbol.size() != 0 && leftSymbol[leftSymbol.size() -1] =='(') ||
(s[i] == ']' && leftSymbol.size() != 0 && leftSymbol[leftSymbol.size() -1] =='[') ||
(s[i] == '}' && leftSymbol.size() != 0 && leftSymbol[leftSymbol.size() -1] =='{')
          ){
            leftSymbol.pop_back();
          }else{
              return false;
          }
        
    }
        return !leftSymbol.size();
};
             };
             